enclave-tls/src/enclave_quotes/sgx-ecdsa/ecalls.c

```diff
@@ -14,11 +14,12 @@
 
 sgx_status_t ecall_generate_evidence(uint8_t *hash, sgx_report_t *app_report)
 {
-	sgx_report_data_t report_data = { 0, };
+	sgx_report_data_t report_data;
 	assert(sizeof(report_data.d) > SHA256_HASH_SIZE);
 	memcpy(report_data.d, hash, SHA256_HASH_SIZE);
```

Comment:
If the warning is caused by assert(), we should use build time assert, e.g, BUILD_BUG_ON(), instead of runtime assert.

Reply:
The build warning is caused by sgx_report_data_t report_data = { 0, };  and sgx_target_info_t qe_target_info = { 0 };

Reply:
Use memset() to zero the report_data. If the size of report_data is less than the size of hash, e.g, sha1, the trailing portion of report_data is not initialized.

Reply:
ok. changed